src/storeforward/Module.md

```diff
@@ -0,0 +1,375 @@
+    if(payload is json) {
+        log:printInfo("Response received " + "Response status code= "+ resp.statusCode + ": "+ payload.toString());
+    } else {
+        log:printError("Error while getting response payload ", err=payload);
+    }
+}
+var myMessageProcessor = new storeforward:MessageForwardingProcessor(myProcessorConfig, handleResponseFromBE);
+...
+```
+
+>Note The time taken to process the response message will get added to the message forwarding interval. 
+
+## Integration patterns 
+
+### 1. Reliable delivery
+
+The connector can be used to forward an HTTP message reliable to a HTTP service. If you need to expose an unreliable HTTP service
+in an reliable manner, this connector can be used. Message will not get discarded until it is successfully delivered to the service. 
+The poison messages which cannot be processed by the service, can be routed to DLC store and users can manually deal with them. 
+
+### 2. In order message delivery 
+
+This connector stores messages in a queue of a message broker, which keeps FIFO behavior. Hence, the inbound messages are stored
+in the order they reached the service. When polling, messages are picked and processed once after the other, where the order
+of messages in the queue is kept when delivering to the backend. If a message is failed to process, the processor can be configured
+to stop until issue is fixed, so that the order is ensured.
+
+### 3. Message throttling 
+
+Sometimes, there are legacy backend services in the systems, which cannot service inbound requests in a high throughput. they may crash
+or go out of resources. If there are spikes in the inbound request rate, which backend service cannot withhold, there is a need to regulate
+the message rate. 
+
+Messages can be stored in the incoming rate. Message forwarding rate is governed by following parameters. 
+
+* pollTimeConfig - users can configure an interval where forwarding task triggers. Even a cron expression is possible. (i.e `0/2 * * * * ?`)
+* forwardingInterval - By default this is configured to`0`. It is effective only when `batchSize > 1`. The messages in a 
+                       batch will be forwarded to the HTTP service with a delay of `forwardingInterval` between the messages configured in milliseconds. By default `batchSize = 1`. 
+
+### 4. Asynchronous messaging 
+
+By design, messages are not processed in a synchronous manner. Inbound messages are stored and then they are processed. Processing can be scheduled 
+so that messages came in are processed later in off-peak hours. Storing service and processing service can be two separate containers. 
+These are the semantics of asynchronous message processing. Reliable delivery is added on top of it. Thus, in places where asynchronous messaging
+is required, this connector can be used. 
+
+## Extensibility
+
+As the connector is using `JMS` connector provided by Ballerina, any message broker with the support for `JMS` can be configured
```

Comment:
Shall we remove the part saying provided by Ballerina?

Reply:
Fixed.